### ISSUE TITLE
Unlocking

### ISSUE THREAD

**Issue by Aeitto** [sl4v3k/Shapr3d_crack#2]
Sławek możesz powiedzieć o co tu chodzi?
Trzeba mieć IDE czy co?

**Comment by sl4v3k** [comment]
Hej, nie nie musisz miec IDY, mozesz to zrobic na dwa sposoby:
1. bezposredni spaczowac binarke
2. spaczowac w pamieci uzywajac debugera np x64

daj mi dostep zdalny to pokaze Ci jak to zrobic drugim sposobem

**Comment by Aeitto** [comment]
Nie znam się na tym a chciałbym się nauczyć.
Każden program ma zawsze kluczową funkcje za paywallem, czy to fusion czy
inne.

Nie wiem gdzie zacząć. Może jakiś poradnik uniwersalny

**Comment by dwg93** [comment]
hello, I am a new entry and I have no experience, I try to find the address 00000001815CEDAB in my shapr3d.dll
but there isn't. probably it isn't the same file. Can sl4v3k help me with the file attached?
[Shapr3D.zip](https://github.com/sl4v3k/Shapr3d_crack/files/11304337/Shapr3D.zip)

**Comment by sl4v3k** [comment]
Your dll comes from different version, so correct address is: 000000018177CB4F
decompiled code:
.text:000000018177CB47 4C 8D 45 01                             lea     r8, [rbp+70h+var_6F]
.text:000000018177CB4B 48 8B 50 18                             mov     rdx, [rax+18h]
replace this call by: mov rax,rcx (48 8B C1 90 90)
.text:**000000018177CB4F** E8 1C E3 86 FF                          call    sub_180FEAE70
.text:000000018177CB54 48 8B D0                                mov     rdx, rax
.text:000000018177CB57 48 8B CE                                mov     rcx, rsi
.text:000000018177CB5A E8 71 E2 FF FF                          call    sub_18177ADD0

**Comment by dwg93** [comment]
thanks, I will try ,and I hope to be able to do your indications.

**Comment by dwg93** [comment]
Hello,I am very stupid but I am not able to go on. Which is the hex editor that you use to see - .text:000000018177CB4B 48 8B 50 18 mov rdx, [rax+18h]- ? there is some tutorial that explains? Pherhaps If you can send to my address dwg93@email.com the file with your modification  I can study around your work and pherhaps learn something.(sorry for my bad english)
best regards

**Comment by sl4v3k** [comment]
i did send patched dll You sent me before...check email

**Comment by sl4v3k** [comment]
by the way, You have to patch address **000000018177CB4F E8 1C E3 86 FF call sub_180FEAE70**
NOT 000000018177CB4B 48 8B 50 18 mov rdx, [rax+18h]

decompilation comes from IDA, its not hex editor output

**Comment by talinedali** [comment]
Dear sl4v3k! I have no experience in hex or dll editing. If you could reach out and tell me which program I should use, or help me set the values yourself in the .dll, that would be great. Please, specify where I could write you

**Comment by sl4v3k** [comment]
You have everything in description -> dll address, values its just 5 bytes to change

**Comment by talinedali** [comment]
While I understand your frustration, please have patience with me. My copy of shapr was downloaded through MS. In shapr3D.dll (opened in HxD) I could not locate that address - maximum offset in 01A2FDF0. Have I missed a step, or chose a wrong program?

**Comment by sl4v3k** [comment]
send me dll from Your version of Shapr

**Comment by talinedali** [comment]
Sadly, I have no idea how to contact you on github. Could you please write me an email at contactmehere@rambler.ru ?

**Comment by sl4v3k** [comment]
Just let me know here what version of shapr You have or send dll to google drive and give link here

**Comment by Franchouillard-surf** [comment]
Hi, I download the Shapr3D.zip from the comment of dwg93 upside to get the Shapr3D.dll, i also download HxH Hex Editor and open the .dll
I try everything in the comment but i could not find "00000001815CEDAB" or "000000018177CB4F"...
The maximum offset is 01A2FDF0
Could you help me please ?

Thanks a lot !

**Comment by sl4v3k** [comment]
he addresses i am providing are from IDA decompilation, in order to locate it in dll bin file using hex editor is to search for same bytes as listed next to addresses

**Comment by sl4v3k** [comment]
Just let me know here what version of shapr You have or send dll to google drive and give link here, i will look for right offset

**Comment by Franchouillard-surf** [comment]
gad damn i'm dumb, i was looking for Shape3D X, not Shapr3D as software haha... sorry and thanks anyway !!

**Comment by fortpen** [comment]
Hello. downloaded version shapr3d 5.352.5062.0 (file shapr3d.dll on google drive link https://drive.google.com/drive/folders/1hdPzc62bHE1t_asVuCdHAJARzQf1XKlI?usp=sharing ). I ask you to refer to cured (complete) learner shapr3d.dll

Thank you very much in advance.

**Comment by Franchouillard-surf** [comment]
Sorry i was thinking about shape3d and not shapr3d, not thé same
software....
My mistake!!!
Thanks anyway !!

**Comment by fortpen** [comment]
> give me upload privileges i will put file there back, or go to offset 0x178989F (E8 1C AE 86 FF) change to (48 8B C1 90 90)

provided.

https://drive.google.com/drive/folders/1hdPzc62bHE1t_asVuCdHAJARzQf1XKlI?usp=sharing

**Comment by sl4v3k** [comment]
done, please test and let me know

**Comment by fortpen** [comment]
> done, please test and let me know

file replaced. The Shapr3D program does not start. It asks to check for an error in the MS Store. Then the system in the background overwrote the modified Shapr3d.dll file with the factory one.  Then Shapr3d.exe starts with trial restrictions.

**Comment by sl4v3k** [comment]
Yes, MS Store directory is protected place,  it is necessary to make additional steps to bypass

**Comment by Aeitto** [comment]
So whats the current working wersion outside MS. We have dll?

**Comment by TopBlogger** [comment]
Hello, is there a similar method in mac os?

**Comment by Aeitto** [comment]
> Yes, MS Store directory is protected place, it is necessary to make additional steps to bypass

Can you tell how?
Methods form here dosnt work:
https://superuser.com/questions/940693/editing-creating-files-in-c-program-files-windowsapps

**Comment by werewolffox** [comment]
Hi, can you patch my dll file

**Comment by manyambitions** [comment]
Ja też chcę być hacker : ) manyambitions@protonmail
Shapr3D.Shapr3D_5.420.5479.0
https://tmpfiles.org/dl/1978184/shapr3d.zip 
Proponuję portable version, z racji problemów z uprawnieniami, nowymi wersjami i update.

**Comment by lmpk1** [comment]
Hello!
I can't find stroke in .dll file. Ver [5.493.5987.0 #f98b370b] . Use Hex editor neo.
Could you help me, please?
https://drive.google.com/file/d/18jB13VjyD2cfgDME5FQhcITmSjG82alf/view?usp=drive_link
Thnks!
